src/main/java/com/zuniorteam/lotto/core/Prize.java

```diff
@@ -1,51 +1,59 @@
 package com.zuniorteam.lotto.core;
 
 
-import java.util.Collections;
-import java.util.Map;
+import com.zuniorteam.lotto.vo.PrizeRule;
+import com.zuniorteam.lotto.vo.PrizeRule.BonusMatch;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-public enum Prize {
-
-    ZERO(0, 0L),
-    ONE(1, 0L),
-    TWO(2, 0L),
-    THREE(3, 5000L),
-    FOUR(4, 50000L),
-    FIVE(5, 1500000L),
-    SIX(6, 20000000000L);
+import static com.zuniorteam.lotto.vo.PrizeRule.BonusMatch.*;
 
-    private static final Map<Integer, Prize> MATCH_COUNT_TO_PRIZE;
+public enum Prize {
 
-    static {
-        final Map<Integer, Prize> matchCountToPrize = Stream.of(values())
-                .collect(Collectors.toMap(Prize::getMatchCount, e -> e));
+    WINNER(1, new PrizeRule(6, ANYWAY), 20000000000L),
+    BONUS_WINNER(2, new PrizeRule(5, MATCH), 30000000L),
+    THIRD_PRIZE(3, new PrizeRule(5, NO_MATCH), 1500000L),
+    FOURTH_PRIZE(4, new PrizeRule(4, ANYWAY), 50000L),
+    FIFTH_PRIZE(5, new PrizeRule(3, ANYWAY), 5000L),
+    SIXTH_PRIZE(6, new PrizeRule(2, ANYWAY), 0L),
+    SEVENTH_PRIZE(7, new PrizeRule(1, ANYWAY), 0L),
+    LOSER(8, new PrizeRule(0, ANYWAY), 0L);
+
+    private final int rank;
+    private final PrizeRule prizeRule;
+    private final long money;
 
-        MATCH_COUNT_TO_PRIZE = Collections.unmodifiableMap(matchCountToPrize);
+    Prize(int rank, PrizeRule prizeRule, long money) {
+        this.rank = rank;
+        this.prizeRule = prizeRule;
+        this.money = money;
     }
 
-    private final int matchCount;
-    private final long money;
+    public long getMoney() {
+        return money;
+    }
 
-    Prize(int matchCount, long money) {
-        this.matchCount = matchCount;
-        this.money = money;
+    public int getRank() {
+        return rank;
     }
 
     public int getMatchCount() {
-        return matchCount;
+        return prizeRule.getMatchCount();
     }
 
-    public long getMoney() {
-        return money;
+    public boolean hasBonus() {
+        return prizeRule.hasBonus();
     }
 
-    public static Prize ofByMatchCount(Integer matchCount) {
-        if (!MATCH_COUNT_TO_PRIZE.containsKey(matchCount)) {
-            throw new IllegalArgumentException("적절하지 않은 매칭 갯수입니다.");
-        }
-
-        return MATCH_COUNT_TO_PRIZE.get(matchCount);
+    public static Prize ofByMatchCountAndBonus(int matchCount, boolean hasBonusNumber) {
```

Comment:
```suggestion
    public static Prize createByMatchCountAndBonus(int matchCount, boolean hasBonusNumber) {
```
전치사가 두개 연달아 붙는게 어색해서...슬쩍 제안해봅니다.